homeassistant/components/escea/climate.py

```diff
@@ -0,0 +1,299 @@
+        self._available = True
+
+        self._device_info = {
+            "identifiers": {(ESCEA, self.unique_id)},
+            "name": self.name,
+            "manufacturer": "Escea",
+        }
+
+    async def async_added_to_hass(self):
+        """Call on adding to hass."""
+        # Register for connect/disconnect/update events
+        @callback
+        def controller_disconnected(ctrl: Controller, ex: Exception) -> None:
+            """Disconnected from controller."""
+            if ctrl is not self._controller:
+                return
+            self.set_available(False, ex)
+
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass, DISPATCH_CONTROLLER_DISCONNECTED, controller_disconnected
+            )
+        )
+
+        @callback
+        def controller_reconnected(ctrl: Controller) -> None:
+            """Reconnected to controller."""
+            if ctrl is not self._controller:
+                return
+            self.set_available(True)
+
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass, DISPATCH_CONTROLLER_RECONNECTED, controller_reconnected
+            )
+        )
+
+        @callback
+        def controller_update(ctrl: Controller) -> None:
+            """Handle controller data updates."""
+            if ctrl is not self._controller:
+                return
+            self.async_write_ha_state()
+
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass, DISPATCH_CONTROLLER_UPDATE, controller_update
+            )
+        )
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._available
+
+    @callback
+    def set_available(self, available: bool, ex: Exception = None) -> None:
+        """Set availability for the controller."""
+        if self.available == available:
+            return
+
+        if available:
+            _LOGGER.info("Reconnected controller %s ", self._controller.device_uid)
+        else:
+            _LOGGER.info(
+                "Controller %s disconnected due to exception: %s",
+                self._controller.device_uid,
+                ex,
+            )
+
+        self._available = available
+        self.async_write_ha_state()
+
+    @property
+    def device_info(self):
+        """Return the device info for the Escea system."""
+        return self._device_info
+
+    @property
+    def unique_id(self):
+        """Return the ID of the controller device."""
+        return self._controller.device_uid
+
+    @property
+    def name(self) -> str:
+        """Return the name of the entity."""
+        return f"Escea Fireplace {self._controller.device_uid}"
+
+    @property
+    def should_poll(self) -> bool:
+        """Return True if entity has to be polled for state.
+
+        False if entity pushes its state to HA.
+        """
+        return False
```

Comment:
Use `_attr_should_poll`, see also the comment below on `_attr_*`.

Reply:
Done, thanks very much for the whole review.